### PR TITLE
John's Incredibly Spiteful Admin Balldance

### PR DESCRIPTION
**Pull request by Jc1469**
:cl:  
bugfix: AO now can't view a lot of shit.
/:cl:

### CODE REVIEW

**Comment by Slinky5890** [comment]
One day AO will become just an OOC tag

**Comment by Partheo** [comment]
AOsquad dream is dead.

**Comment by Alexandria404** [comment]
The AO hordes shall grow, nothing can stop them now.

**Comment by Kmc2000** [comment]
-1 your last balldance removed SeniorMin.

**Comment by Slinky5890** [comment]
I think matsku was the one that removed seniormin

File: code/modules/admin/admin.dm [hunk 1]
```diff
@@ -2,23 +2,30 @@
...
+	var/list/validmins = list()
+	if(fuckholder)
+		for(var/client/C in admins)
+			if(check_rights_for(C, R_BASIC))
+				validmins += C
+		msg = "<span class=\"admin\"><span class=\"prefix\">ADMIN LOG:</span> <span class=\"message\">[msg]</span> - MESSAGE NOT SENT TO PEOPLE WITHOUT +BASIC</span>"
+		validmins << msg
+	else
+		msg = "<span class=\"admin\"><span class=\"prefix\">ADMIN LOG:</span> <span class=\"message\">[msg]</span></span>"
+		admins << msg
```
Comment (ghost) [inline code/modules/admin/admin.dm]: to_chat

File: code/modules/admin/admin.dm [hunk 2]
```diff
@@ -2,23 +2,30 @@
...
-	msg = "<span class=\"admin\"><span class=\"prefix\">ADMIN LOG:</span> <span class=\"message\">[msg]</span></span>"
-	admins << msg
+/proc/message_admins(msg, fuckholder = 0)
+	var/list/validmins = list()
+	if(fuckholder)
+		for(var/client/C in admins)
+			if(check_rights_for(C, R_BASIC))
+				validmins += C
+		msg = "<span class=\"admin\"><span class=\"prefix\">ADMIN LOG:</span> <span class=\"message\">[msg]</span> - MESSAGE NOT SENT TO PEOPLE WITHOUT +BASIC</span>"
+		validmins << msg
```
Comment (ghost) [inline code/modules/admin/admin.dm]: to_chat

**Review by ghost — CHANGES_REQUESTED** [review]

File: code/modules/admin/admin.dm [hunk 3]
```diff
@@ -2,23 +2,30 @@
...
-/proc/message_admins(msg)
-	msg = "<span class=\"admin\"><span class=\"prefix\">ADMIN LOG:</span> <span class=\"message\">[msg]</span></span>"
-	to_chat(admins, msg)
+/proc/message_admins(msg, fuckholder = 0)
+	var/list/validmins = list()
+	if(fuckholder)
+		for(var/client/C in admins)
+			if(check_rights_for(C, R_BASIC))
+				validmins += C
+		msg = "<span class=\"admin\"><span class=\"prefix\">ADMIN LOG:</span> <span class=\"message\">[msg]</span> - MESSAGE NOT SENT TO PEOPLE WITHOUT +BASIC</span>"
```
Comment (AshCorr) [inline code/modules/admin/admin.dm]: You dont need to yell`- MESSAGE NOT SENT TO PEOPLE WITHOUT +BASIC` to the world

**Pull request merged**